util/gnmi.go

```diff
@@ -134,6 +134,41 @@ func FindPathElemPrefix(paths []*gpb.Path) *gpb.Path {
 	}
 }
 
+// pathElemMatchesQuery returns true if the given concrete (no wildcard)
+// PathElem is a possible gNMI telemetry name match for the reference PathElem.
+// In particular, it matches wildcards for names and list keys, but *not* "...".
+// See https://github.com/openconfig/reference/blob/master/rpc/gnmi/gnmi-path-conventions.md
+func pathElemMatchesQuery(elem, refElem *gpb.PathElem) bool {
```

Comment:
I'd suggest to call the first argument here `queryElem` and the second `pathElem` so that it's clear which of the two we're trying to match against each other.